runtime/win32.c

```diff
@@ -1113,54 +1104,12 @@ void caml_init_os_params(void)
   QueryPerformanceFrequency(&frequency);
   /* Convert the frequency to the duration of 1 tick in ns */
   frequency.QuadPart = 1000000000LL / frequency.QuadPart;
-
-  /* Get the current time as accurately as we can.
-     GetSystemTimePreciseAsFileTime is available on Windows 8 / Server 2012+ and
-     gives <1us precision. For Windows 7 and earlier, which is only accurate to
-     10-100ms. */
-  pGetSystemTime =
-    (LPFN_GETSYSTEMTIME)GetProcAddress(GetModuleHandle(L"kernel32"),
-                                       "GetSystemTimePreciseAsFileTime");
-  if (!pGetSystemTime)
-    pGetSystemTime = GetSystemTimeAsFileTime;
-
-  /* Get the time and the performance counter. Get the performance counter first
-     to ensure no quantum effects */
-  QueryPerformanceCounter(&counter);
-  pGetSystemTime(&stamp);
-
-  now.LowPart = stamp.dwLowDateTime;
-  now.HighPart = stamp.dwHighDateTime;
-
-  /* Convert a FILETIME in 100ns ticks since 1 January 1601 to
-     ns since 1 Jan 1970. */
-  clock_offset.QuadPart =
-    ((now.QuadPart - 0x19DB1DED53E8000ULL) * 100);
-
-  /* Get the offset between QueryPerformanceCounter and
-     GetSystemTimePreciseAsFileTime in order to return a true timestamp, rather
-     than just a monotonic time source */
-  clock_offset.QuadPart -= (counter.QuadPart * frequency.QuadPart);
-
-  GetSystemTimePreciseAsFileTime(&stamp);
-  now.LowPart = stamp.dwLowDateTime;
-  now.HighPart = stamp.dwHighDateTime;
-  now.QuadPart *= 100;
```

Comment:
I'll bite: what was all this old code doing, and why is it no longer necessary?

Reply:
The last four lines are a rebasing artefact - at the time the function was originally written, losing Windows 7 support was likely to matter (hence the use of `GetProcAddress`). The rest of it was because at some point between 2018-2021 there was a need for the clock source to be both monotonic and a timestamp - that need had disappeared before upstreaming, but the code here wasn't deleted.